server/models/Tier.js

```diff
@@ -378,6 +379,24 @@ export default function (Sequelize, DataTypes) {
     return this.availableQuantity().then(available => available - quantityNeeded >= 0);
   };
 
+  Tier.prototype.setCurrency = async function (currency) {
+    // Nothing to do
+    if (currency === this.currency) {
+      return this;
+    }
+
+    // Check if Orders
+    const count = await models.Order.count({ where: { TierId: this.id } });
+    if (count > 0) {
+      logger.warning(
+        `Can't change currency of tier ${this.id}: ${this.currency} -> ${currency} (total orders: ${count})`,
+      );
+      return this;
+    }
```

Comment:
This should probably return an error, since it means the function failed to complete